### DISCUSSION TITLE
How to access a Google Drive Directory from PC using pcmanfm-qt?

### DISCUSSION THREAD

**Discussion by rajibando** [lxqt/lxqt#2642] [Q&A] [answered]
Hello Team leaders, maintainers and programmers,

On the pcmanfm-qt path what should I type so that I could access a particular shared directory on the Google Drive? Say the Drive's unique link is:
https://drive.google.com/drive/u/0/folders/A_Long_Unique_Code
What should the codeline be?

Should I have to install the package `google-drive-ocamlfuse` from the repository `ppa:alessandro-strada/ppa` ? Or do I have to build from source?: 
Like, Install dependencies: sudo apt install opam fuse libcurl4-gnutls-dev libfuse-dev libsqlite3-dev

Then Initialize OPAM and install: `google-drive-ocamlfuse`?
Please advise.

**Comment by tsujan** [comment, marked as answer]
I use Google Drive with pcmanfm-qt on a regular basis. There may be various ways. This is my setup with `rclone`, which doesn't need `gvfs-google`:

 * I've set up `rclone` once for all. `rclone` has a comprehensive documentation.
 * I've created the directory `~/Downloads/googleDrive`.
 * I've made a [custom action](https://github.com/lxqt/pcmanfm-qt/wiki/custom_actions) like this:

```
[Desktop Entry]
Type=Action
Icon=googledrive
Name=Google Drive
Profiles=profile-zero;

[X-Action-Profile profile-zero]
SelectionCount==1
MimeTypes=inode/directory;
Basenames=googleDrive;
Folders=/home/pedram/Downloads;
Exec=rclone-mine
Name=Default profile
```
It calls `rclone-mine`, which is a simple script that mounts Google Drive (consult `rclone` doc for the command).

With this setup, I just right click the directory `~/Downloads/googleDrive`, click the custom action `Google Drive`, and it'll be mounted and shown in the side-pane. Unmounting can be done in the side-pane too.

![google_drive](https://github.com/user-attachments/assets/510bda71-71bb-420e-821b-7b9c30b78f9a)

EDIT:

The mount command inside my script is like this:

```sh
rclone mount NAME_OF_DRIVE: PATH_TO_DIR --drive-acknowledge-abuse &
```

**Reply by rajibando** [reply]
Before even following your advice I must thank you for replying to my post promptly with an apparently ready solution. Please accept my admiration, Mr. Tsujan.

Were I to access the Google Drive Read-Only, what change would I need to incorporate?

Would I need to alter the script line with the following?:
`rclone mount NAME_OF_DRIVE: PATH_TO_DIR --drive-acknowledge-abuse --read-only &`

However, if the permission to access the Drive is `Share Only` may be your line won't need to be altered.

Thank you once again for replying to my post so promptly.

**Reply by tsujan** [reply]
> Were I to access the Google Drive Read-Only, what change would I need to incorporate?

Sorry, I'm not sure, because I access it in the read-and-write mode, like any other folder in my home.

BTW, I added `--drive-acknowledge-abuse` because, at some point, Google didn't allow me to move/copy files to other folders; it's a workaround by `rclone`.

**Reply by rajibando** [reply]
In that case may my question be ignored. If an owner of the Drive folder doesn't grant  to me Read-Write persmission for the said Folder then the code line won't affect the files in the folder even when I try to alter them.
I didn't find time to experiment as I am learning a new tech, on which I will shortly post a question. I will shortly get back on to this topic. So please bear with me.

**Reply by tsujan** [reply]
OK, found it; see https://rclone.org/drive/ → `scope>`. I used that documentation to set up my Google Drive and chose the default options in most cases. The doc is very easy to follow.
